### ISSUE TITLE
seek funding from Mozilla

### ISSUE THREAD

**Issue by whit537** [gratipay/inside.gratipay.com#637]
In conversation at #314, [Louis-David Benyayer](https://twitter.com/LDBenyayer/status/735556364269916162) suggested that we seek funding from Mozilla and other large open-source foundations.

https://twitter.com/LDBenyayer/status/735556364269916162

---

We're writing our proposal at https://public.etherpad-mozilla.org/p/gratipay-moss-track-2-2016.

The application form is at https://docs.google.com/forms/d/1rwYQTT-9-eldS-kElY646bMwMzJpxfL8lDskX86xgCQ/viewform.

**Comment by whit537** [comment]
https://www.mozilla.org/en-US/grants/

**Comment by whit537** [comment]
https://wiki.mozilla.org/MOSS/Projects_in_use_by_Mozilla

**Comment by whit537** [comment]
https://blog.mozilla.org/blog/2015/10/23/mozilla-launches-open-source-support-program/
https://wiki.mozilla.org/MOSS

**Comment by whit537** [comment]
Looks like there are three main programs, and the one we'd fit best is [Mission Partners](https://wiki.mozilla.org/MOSS/Mission_Partners)—which has a deadline in a few days!

> The deadline for applications for the first round of awards is 23:59 on Tuesday 31st May 2016.

**Comment by whit537** [comment]
To: [Mozillian introduced in private email by LDB]
Subject: Gratipay & Mozilla

> Thank you for the introduction, Louis-David! Moving you to bcc ...
>
Greetings, []! I'm not sure what Louis-David has shared with you already. I work on a project called [Gratipay](https://gratipay.com/), which provides payments for open companies. We launched four years ago and have moved a little over 1M USD. We are funded via pay-what-you-want (prix libre) on our own platform, and we operate as openly as we responsibly can.
>
So far we have been bootstrapping from revenue, but Louis-David and I had a conversation during OuiShare Fest in which he suggested that Mozilla might be a good partner to help capitalize Gratipay further. I've started looking into Mozilla's funding initiatives on [this public GitHub ticket](https://github.com/gratipay/inside.gratipay.com/issues/637). Are you a good person to talk to further about this? I note that the Mission Partners program has an initial deadline of May 31. Do either of these times work for you for a video call?
>
Friday, May 27 (tomorrow) at 15h00 Paris time
Monday, May 30 at 15h00
>
My preference would be to use Google Hangouts on Air so that we can live-stream and archive the call on YouTube. Skype or another private method would be fine as well, though I'll at least need to summarize our conversation publicly for the sake of the Gratipay community.
>
Thanks for your time! :-)

**Comment by whit537** [comment]
Ping @bbangert @tarekziade.

Do you think Gratipay is a good candidate for the Mozilla MOSS Mission Partners track?

**Comment by whit537** [comment]
> Hmm ... maybe @Gratipay should apply? Discussing here: https://github.com/gratipay/inside.gratipay.com/issues/637 …. cc: @AlexSalkever @tarek_ziade

https://twitter.com/whit537/status/735841509501751296

**Comment by whit537** [comment]
Looks like the Mission Partners track was launched a couple weeks ago:

https://groups.google.com/forum/#!topic/mozilla.moss/kEfcZkfb2Go
https://blog.mozilla.org/blog/2016/05/11/mozilla-open-source-support-moss-now-open-to-all-projects/
https://twitter.com/mozilla/status/730421316432613376

**Comment by whit537** [comment]
> Therefore, applications for Mission Partners do not require a Mozillian to support them, as applications for Foundational Technology do.

https://wiki.mozilla.org/MOSS/Mission_Partners#FAQs

**Comment by whit537** [comment]
> Instead, they must be endorsed by a well-known and respected figure from a wider software community of which the project is a part, who is not directly connected with your project. This could be a language community, or a national community - it does not have to be the international open source community.

Finding an endorser within the deadline seems like a big part of the work here. Who could we ask?

 - Guido?
 - Resig?
 - Kenneth?

**Comment by whit537** [comment]
The other interesting questions are:

**Requested amount:**

_Please give in US dollars (max: $250,000)_

**What are the concrete, specific outputs and outcomes this grant would produce, and how do those activities further the Mozilla Mission?**

_Please describe what you would use the funds for - what you are going to build, hack or fix. Also, explains how it furthers our mission, perhaps tying your activities to items in the Manifesto - https://www.mozilla.org/about/manifesto/. (Max 8k chars)_

**Please tell us more about how your project is managed. Please describe your core team.**

**Please tell us more about your community.**

**Comment by whit537** [comment]
> Finding an endorser within the deadline seems like a big part of the work here. Who could we ask?

Maybe Jeff Spies?

**Comment by whit537** [comment]
> I skimmed through GratiPay's site.
I'm not the right person to talk to about business, I have an engineering role here. I also host lots of events in our space (in Paris).
And I don't know personally the right people in Mozilla (I know their team and names though).
So if you found information on how to get funding from us, you know more than I do.
>
I'm not sure it's useful to record our conversation as there will be nothing official I can tell you.
But if you still want some informal contact (who knows if I have an important info for you), let me know and I'll get on hangouts by 15:00 tomorrow.
>
Cheers, and maybe tomorrow !

----------------

Thanks, []. Let's hold off on a call for now. The MOSS pages list some potential contacts, I think I will follow up on those.

Thanks for the response and best wishes for now! À plus tard ! :-)

**Comment by whit537** [comment]
# Draft 1

**Project description:**

Gratipay offers payment products for open organizations: pay-what-you-want payments from customers to organizations, and take-what-you-want compensation from open organizations to their collaborators. Our customers include open-source projects, open SaaS products, and coworking spaces. We've been around for four years and have moved over $1 million (US), though we had to reboot the business a year ago due to legal concerns. Now we're approaching 200 customers on Gratipay 2.0, and we move $1,000 per week. Gratipay competes on price because we're pay-what-you-want on our own platform (our "soft fee" ends up being 5% of total volume), and we compete on mission because we're here to cultivate an economy of gratitude, generosity, and love! :-)

----------

**Requested amount:**

_Please give in US dollars (max: $250,000)_

$250,000

----------

**What are the concrete, specific outputs and outcomes this grant would produce, and how do those activities further the Mozilla Mission?**

_Please describe what you would use the funds for - what you are going to build, hack or fix. Also, explains how it furthers our mission, perhaps tying your activities to items in the Manifesto - https://www.mozilla.org/about/manifesto/. (Max 8k chars)_

== Outcomes ==

This grant would fund two orders of magnitude of growth for Gratipay. As mentioned, we currently move $1,000 per week for ~200 customers. Our goal with this grant would be to reach $100,000 weekly volume for ~20,000 customers within two years. The grant would fund activities of the Gratipay core team, including but not limited to marketing, sales, product design and development, customer support, accounting, security, compliance, and governance. Here are specific projects on our plate:

Improve the product.—There are many glaring deficiencies in our basic giving workflow for individual and corporate givers, and on the receiver side as well.

Reimplement payouts.—Our main innovation is take-what-you-want compensation, but we turned this off after a two-year pilot when we rebooted as Gratipay 2.0. We need to bring it back.

Implement identity verification.—We need to store and verify national identity information in order to participate more fully in the global financial system.

Strengthen security.—We need to evolve our security program, especially as we start handling identity information.

Diversify payments infrastructure.—We need to work with more partners to reduce risk and costs, and expand our service.

Account for funds.—We recently set up a double-entry accounting system with a CPA, and now we need to input four years of data.

Clean up tech debt.—In particular, we made some database schema mistakes that are slowing us down in other areas.

For more details, please see our recent "Making it Right" blog post:

https://gratipay.news/making-it-right-f8e1eccb46e

== Mission Alignment ==

Mozilla's mission is "to ensure the Internet is a global public resource, open and accessible to all. An Internet that truly puts people first, where individuals can shape their own experience and are empowered, safe and independent."

Fueling Gratipay's growth furthers Mozilla's mission by contributing to a vibrant and sustainable open-source software ecosystem (Principle 7: "Free and open source software promotes the development of the Internet as a public resource").

Fueling Gratipay's growth protects and empowers individuals on the Internet (Principles 3, 4, and 5), because Gratipay's pay-what-you-want model is less invasive and privacy-eroding than the ad models currently prevalent.

Fueling Gratipay's growth improves the organizations that provide the Internet (Principles 5, 6, and 8), because Gratipay strongly encourages *open* organizations.

Gratipay's values of safety, consent, transparency, openness, and love ("the ladder of love"— http://inside.gratipay.com/big-picture/brand/) strongly align with Mozilla's principles of security and privacy (Principle 4), individual autonomy (Principle 5), and transparency (Principle 8). Indeed, Gratipay has been a leader in developing "[t]ransparent community-based processes [that] promote participation, accountability and trust" (Principle 8). Furthermore, Gratipay aligns with Mozilla's principle of balancing "commercial profit and public benefit" (Principle 9): we exist to make collaboration economically normal.

----------

**Please tell us more about how your project is managed. Please describe your core team.**

Gratipay is managed as a benevolent dictatorship with a core team of about seven people from four countries. Our internal processes are documented on our (public) intranet, /Inside Gratipay/:

http://inside.gratipay.com/big-picture/welcome

----------

**Please tell us more about your community.**

Gratipay has two layers of community: contributors and users. We have had 100+ code contributors and 900+ GitHub issue commenters over the past four years, and 100 people took compensation from Gratipay during our 2-year take-what-you-want pilot. We've held three annual in-person contributor retreats. In our 1.0 phase we saw about 4200 users from 30+ countries, and under Gratipay 2.0 we currently have 187 approved receivers and about 750 givers.

**Comment by whit537** [comment]
To: Resig
Subject: dustin' off the ol' Gratipay

> John!
>
Hey, man. Hope you're well. :-)
>
I'm writing because we're starting to ramp Gratipay back up again after a couple rocky years, and as part of that we're looking to apply for a grant from Mozilla. One of the things they're asking for is an endorsement from "a well-known and respected figure from a wider software community of which the project is a part, who is not directly connected with your project."
>
They're asking for "a paragraph or two on why supporting your work via this grant is a great idea." You wrote such [a nice post](http://ejohn.org/blog/gittip-at-khan-academy/) about Gittip back in the day—we could almost use parts of it verbatim here.
>
Would you be willing and able to endorse Gratipay for this [grant application](https://github.com/gratipay/inside.gratipay.com/issues/637#issuecomment-222000408)? :-)

**Comment by whit537** [comment]
I posted a message on the [mozilla.moss](https://groups.google.com/forum/#!forum/mozilla.moss) Google Group, but it seems to have gotten swallowed—perhaps into a moderator queue? I'll check again later ...

**Comment by whit537** [comment]
Yesssssss ...

https://groups.google.com/forum/#!topic/mozilla.moss/CZY1TiyZOyY

> Some questions which occur to me: 
>
* This seems a little like a request for VC money. Is it? 
>
* Is Gratipay's software open source? I assume it is because that's a stated criteria, but it's not noted anywhere. 
>
* It's not clear how $250,000 will translate into two orders of magnitude of growth (as opposed to, say, one, three or no orders of magnitude). 
>
* It's not clear how $250,000 maps onto the list of tasks that need to be done, other than that it's the maximum amount you are allowed to ask for and there's lots to do. 
>
* It's not clear how "take-what-you-want" works and in what context people would use it. 
>
* The Mission Alignment section makes many statements but doesn't back them up very well. 
>
* Perhaps a wiki page rather than an issue might be a better place to collaboratively or incrementally author a document? :-) 
>
(John Resig would be an entirely appropriate person to be an endorser.) 
>
Gerv

**Comment by whit537** [comment]
I've moved the draft [to an etherpad](https://public.etherpad-mozilla.org/p/gratipay-moss-track-1-2016). I'm going to pursue the conversation over on mozilla.moss without cross-posting here, since mozilla.moss is an open list.

**Comment by whit537** [comment]
-  [ ] Explain how $250,000 will translate into two orders of magnitude of growth (as opposed to, say, one, three or no orders of magnitude).
 - [ ] Explain how $250,000 maps onto the list of tasks that need to be done, other than that it's the maximum amount you are allowed to ask for and there's lots to do.
 - [ ] Explain how "take-what-you-want" works and in what context people would use it.
 - [ ] Back up statements in the Mission Alignment section.

**Comment by whit537** [comment]
> Finding an endorser within the deadline seems like a big part of the work here. Who could we ask?

Maybe mitsuhiko?

**Comment by mattbk** [comment]
Stats for growth:

> Gratipay processes about $1,000 per week for 178 Teams and about 700 ~users.

1 OOM growth: $10,000 per week, 1,780 Teams, 7,000 ~users
2 OOM growth: $100,000 per week, 17,800 Teams, 70,000 ~users
3 OOM growth: $1,000,000 per week, 178,000 Teams, 700,000 ~users

To get that sort of growth, I think we need to address some of the core "financial system" issues we're having and make things easy. How to make taxes easier? How to include as many different payin and payout routes as possible? At the same time, how to scale up Team review _or_ really proxy out the KYC stuff?

**Comment by mattbk** [comment]
This a great exercise to think about growth, anyway. If we're focusing on the long tail and using the processing amount as a growth measure, the number of teams and ~users will probably grow much, much more in order to meet each order of magnitude increase in processing.

**Comment by kaguillera** [comment]
> 
Some questions which occur to me:
- This seems a little like a request for VC money. Is it?
- Is Gratipay's software open source? I assume it is because that's a stated criteria, but it's not noted anywhere.
- It's not clear how $250,000 will translate into two orders of magnitude of growth (as opposed to, say, one, three or no orders of magnitude).
- It's not clear how $250,000 maps onto the list of tasks that need to be done, other than that it's the maximum amount you are allowed to ask for and there's lots to do.
- It's not clear how "take-what-you-want" works and in what context people would use it.
- The Mission Alignment section makes many statements but doesn't back them up very well.
- Perhaps a wiki page rather than an issue might be a better place to collaboratively or incrementally author a document? :-)  
>
(John Resig would be an entirely appropriate person to be an endorser.)  
Gerv


My only concern is how were are supposed to address all of those points in 8k chars.
😱

**Comment by whit537** [comment]
@kaguillera :-) Check out [the thread](https://groups.google.com/forum/#!topic/mozilla.moss/CZY1TiyZOyY). We had some additional back-and-forth.

> To get that sort of growth,

@mattbk I think what Gerv is questioning is how predictable growth is. VCs are in the business of taking risks—counting on 9 out of 10 investments to fail, so number 10 has to *really* win. We have to keep in mind that MOSS isn't VC, it's a grant.

Maybe a middle ground between the "$X for Y feature" and "Zx growth" paradigms would be to put together an annual budget, and use that together with [this tool](http://growth.tlb.org/) (https://github.com/gratipay/inside.gratipay.com/issues/405#issuecomment-195014364) to make some worst, likely, and best-case predictions. Here's [a spreadsheet](https://docs.google.com/spreadsheets/d/1BvG-YfuCYdNqzg4-zkvgs8V8MzHImUlgQuncwzbCJEI/edit?usp=sharing). What numbers should we plug in for you, @aandis @clone1018 @mattbk @rohitpaulk @TheHmadQureshi et al.? What other line-items are missing or should be changed?

**Comment by clone1018** [comment]
I'll take $1/week. Are hosting costs included in operations or is this just "payroll"?

**Comment by whit537** [comment]
http://paulgraham.com/aord.html

Our revenue right now is $200/wk. Our expenses are $100/wk, so technically we're default alive right now but only because we're all willing to work for free, which isn't sustainable and defeats the whole purpose of Gratipay. A few scenarios:

| Growth Rate (%) | Weekly Expenses ($) | Capital Required ($k) | Years to Profitability | 
|:----------------:|:-----------------:|:---------------:|:----------------:|
| 0.00 | 3,000 | ∞ | ∞ |
| 1.25 | 3,000 | 432 | 4.2 |
| 1.25 | 4,000 | 672 | 4.6 |
| 2.50 | 3,000 | 218 | 2.1 |
| 2.50 | 4,000 | 322 | 2.3 |
| 2.50 | 5,000 | 467 | 2.5 |

**Comment by whit537** [comment]
> Are hosting costs included in operations or is this just "payroll"?

Payroll is "Core Team" on that spreadsheet. I'm taking the operations number from [the old finances spreadsheet](https://docs.google.com/spreadsheets/d/1p3DpF9ZLEsViBx0685FwJaYN1vVScKVUgXHb2zyqXZg/pub?output=html&widget=true), which includes (is mostly) hosting costs.

**Comment by whit537** [comment]
Alright, so (re)reading http://paulgraham.com/aord.html, our biggest problem is our 0% growth over the past year. Number of teams is increasin, but users and volume are constant, and revenue is slipping:

![screen shot 2016-05-27 at 2 42 47 pm](https://cloud.githubusercontent.com/assets/134455/15617951/5cf63232-2419-11e6-972f-5cb321442215.png)

![screen shot 2016-05-27 at 2 41 01 pm](https://cloud.githubusercontent.com/assets/134455/15617950/5cf60136-2419-11e6-80d1-152257d1f544.png)

**Comment by whit537** [comment]
That doesn't make us look very good. 😞

**Comment by whit537** [comment]
> What the company should have done is address the fundamental problem: that the product is only moderately appealing. Hiring people is rarely the way to fix that. More often than not it makes it harder. At this early stage, the product needs to evolve more than to be "built out," and that's usually easier with fewer people.
>
Asking whether you're default alive or default dead may save you from this. Maybe the alarm bells it sets off will counteract the forces that push you to overhire. Instead you'll be compelled to seek growth in other ways. For example, by doing things that don't scale, or by redesigning the product in the way only founders can.

http://paulgraham.com/aord.html

**Comment by whit537** [comment]
Gah. I am feeling like applying for a MOSS grant is a waste of time until we can demonstrate some growth. 😞

**Comment by kaguillera** [comment]
I don't agree...The reason (in my opinion) that we have not grown is because of the removal of payroll/payments. This was/is our may features that allow open-source developers and their contributors the ability to work on there project full-time. The same applies to the Gratipay's core team. The core team including you need to be able to focus on the development of the system and this funding would allow that. Thus this Paul Graham may have a point but I don't we are an exception to this case and thus could do with the Grant. Think of how **you** can build Gratipay without having to worry about money or working for someone else and taking your time away from developing your dream.
Again the worst that can happen is the say no.

**Comment by whit537** [comment]
> Maybe mitsuhiko?

To: Armin Ronacher
Subject: endorse Gratipay for Mozilla grant?

> Greetings! Hope you're well. :-)
>
I'm writing because Gratipay is [seeking a grant](https://github.com/gratipay/inside.gratipay.com/issues/637) from Mozilla, and one of the things they're asking for is an endorsement from "a hacker with a reputation." We thought of you! :-) The commitment would be to write "a paragraph or two on why [Mozilla] supporting [Gratipay's] work via this grant is a great idea," and possibly to field follow-up questions from Mozilla.
>
Would you be willing to help us out with this?

**Comment by whit537** [comment]
> I am feeling like applying for a MOSS grant is a waste of time until we can demonstrate some growth.

On the other hand, we're at least stabilized—we're not _shrinking_. :-)

**Comment by whit537** [comment]
I note that PyCon is happening right now. Armin may be away from email.

**Comment by bbangert** [comment]
Whats the gratipocalypse?

**Comment by whit537** [comment]
> Whats the gratipocalypse?

@bbangert The Gratipocalypse is the week we went out of business because the legality of our old terms of service were questionable. Deets at https://gratipay.news/gratipocalypse-42fd0ec0d9e8. We bounced right back but have been flat for a year since while we clean up and get reoriented.

![screen shot 2016-05-27 at 8 07 13 pm](https://cloud.githubusercontent.com/assets/134455/15623774/b231e73c-2446-11e6-9d7e-a49cff7dbc09.png)

https://gratipay.com/about/stats

**Comment by whit537** [comment]
@bbangert Wanna [endorse](https://wiki.mozilla.org/MOSS/Mission_Partners#FAQs) us? 😎

**Comment by whit537** [comment]
Alright, I've reworked the application in light of the conversation with Gerv.

https://public.etherpad-mozilla.org/p/gratipay-moss-track-2-2016

I took the 4% best-case scenario from the spreadsheet at https://github.com/gratipay/inside.gratipay.com/issues/78#issuecomment-51561491 (which is in months; I divided by 4.333 to convert to weeks).

**Comment by whit537** [comment]
Didn't get this in before the first-round deadline (largely for lack of endorsement). Let's keep an eye on the first-round results and consider applying down the line.

**Comment by whit537** [comment]
> The first eight awardees are:
>
> [Tor](https://www.torproject.org/): $152,500.
> [Tails](https://tails.boum.org/): $77,000.
> [Caddy](https://caddyserver.com/): $50,000.
> [Mio](https://github.com/carllerche/mio): $30,000. 
> [DNSSEC/DANE Chain Stapling](https://www.getdnsapi.net/): $25,000.
> [Godot Engine](http://www.godotengine.org/): $20,000.
> [PeARS](http://pearsearch.org/): $15,500.
> [NVDA](http://www.nvaccess.org/): $15,000.

https://blog.mozilla.org/blog/2016/06/22/mozilla-awards-385000-to-open-source-projects-as-part-of-moss-mission-partners-program/

**Comment by whit537** [comment]
Now that it's out, "[Roads and Bridges](www.fordfoundation.org/library/reports-and-studies/roads-and-bridges-the-unseen-labor-behind-our-digital-infrastructure/)" (cf. #732) should be the basis of our case here.

**Comment by whit537** [comment]
The way to approach this is with a small, defined pitch. We're not Tor. $250,000 is laughable. We should come up with something in the $15,000 to $50,000 range. Furthermore, based on the [post](https://blog.mozilla.org/blog/2016/06/22/mozilla-awards-385000-to-open-source-projects-as-part-of-moss-mission-partners-program/) announcing the first mission partners, we need to make it _really really easy_ for them to write a sentence of the form "This award will be used to ______________."

So what should we write the grant for? Looking at our [roadmap](https://github.com/gratipay/inside.gratipay.com/issues/742), we are inching closer to bringing back takes for Team Gratipay. The next thing on our list is to finish cleaning up from the Gratipocalypse. That's not very exciting. Other things that come to mind are design (#298) and travel (#711), though with a deadline tomorrow for #711 that's probably too late (and we're not clear about it anyway). If we're building off of _Roads and Bridges_ then of course it's 3.ii. on the roadmap that we want to pitch: "[Solve Corporate Sponsorship of Open-Source](https://github.com/gratipay/gratipay.com/milestones/Solve%20Corporate%20Sponsorship%20of%20Open-Source)." The problem is that we're not ready for that yet.

Here's what I'm leaning towards:

1. Fold "2.ii. Make Widgets 2.0" in with "3.i. Tighten Onboarding Flow."
1. Pitch Mozilla on a **design refresh**.
 1. Design refresh runs in parallel with the rest of "2. Finish cleaning up from the Gratipocalypse" through the remainder of 2016.
1. Design refresh sets us up to launch into "3. Knock pay-what-you-want out of the park" in 2017.

"This reward will be used to **refresh Gratipay's visual identity**."

Thoughts?

In terms of https://github.com/gratipay/inside.gratipay.com/issues/179#issuecomment-129602076, this would be trying to avoid crowding out by putting money into non-technical work. We could easily spend $50,000 on a redesign, no problem. A year or two ago I referred a fairly straightforward website project that ended up costing $65,000. Compared to the $50/wk that we're talking about paying ourselves (#747), there's real potential for you/me/us to feel resentful. Would it be better to pitch something that involves the money going through team takes? Could we find designers to join our Team Gratipay rather than hiring an agency?

**Comment by whit537** [comment]
In private email (https://github.com/gratipay/inside.gratipay.com/issues/732#issuecomment-236776740), Nadia Eghbal declined to endorse us, because of general reservations about crowdfunding as a sustainable solution.

**Comment by sheerun** [comment]
Is etherpad unavailable only for me?

**Comment by mattbk** [comment]
@sheerun I can't access it either, I get "Service Unavailable."

**Comment by whit537** [comment]
> Is etherpad unavailable only for me?
> I can't access it either, I get "Service Unavailable."

It's back now for me:

https://public.etherpad-mozilla.org/p/gratipay-moss-track-2-2016

**Comment by whit537** [comment]
Okay, I've rewritten the grant based on https://github.com/gratipay/inside.gratipay.com/issues/637#issuecomment-236044779. I ended up pitching it in terms of "improving the new customer sign-up process" (aka #432).

https://public.etherpad-mozilla.org/p/gratipay-moss-track-2-2016

Now to find an endorser ...

**Comment by whit537** [comment]
Probably the thing to do is to go through our current teams and reach out to some of the folks from that list, explain that we're trying to jump-start Gratipay and ask for their help. We want "a well-known and respected figure from a wider software community of which the project is a part, who is not directly connected with your project," so let's maybe target the more well-known projects we've managed to attract.

**Comment by whit537** [comment]
Here's an etherpad with our teams on it:

https://public.etherpad-mozilla.org/p/gratipay-moss-track-2-2016-endorser

I'm going to go through that and whittle down a list of higher-profile projects to approach in private email.

**Comment by whit537** [comment]
Actually, scratch that, here [a CSV of GitHub follower counts](https://gist.github.com/whit537/f2f2977fd8b7ae07ace71b359be61cb1) for Gratipay team owners.

```
[gratipay] $ GH_USER=whit537 GH_PASS=deadbeef ./owners-by-github-followers.py > owners.csv
```

```python
#!/usr/bin/env python
from __future__ import absolute_import, division, print_function, unicode_literals

import csv
import os
import requests
import sys


teams = open('teams.txt').read().split()
gratipay = 'https://gratipay.com/'
gratipay_public = gratipay + '{}/public.json'
github_user = 'https://api.github.com/users/{}'
out = csv.writer(sys.stdout)
USER = os.environ['GH_USER']
PASS = os.environ['GH_PASS']


for team in teams:
    owner = requests.get(gratipay_public.format(team)).json()['owner']
    elsewheres = requests.get(gratipay_public.format(owner)).json()['elsewhere']
    github_username = elsewheres.get('github', {}).get('user_name', '')
    github_url = ''
    nfollowers = -1
    if github_username:
        github_url = 'https://github.com/' + github_username
        nfollowers = requests.get( github_user.format(github_username)
                                 , auth=(USER, PASS)
                                  ).json()['followers']
    out.writerow((gratipay+team+'/', gratipay+owner+'/', github_url, nfollowers))
```

**Comment by whit537** [comment]
Here's the same thing in a Google Sheet for [easy sorting](https://docs.google.com/spreadsheets/d/1SAX0yzwbIjvgpUuvbkNXIhxDMwGzIqOmxIHi67TyJu0/edit?usp=sharing).

**Comment by whit537** [comment]
Looks like we have nine Team ~owners who have more than 1,000 followers on GitHub:

| n | nfollowers | ~owner | notes |
|---:|-------------:|-----------|--------|
| 1 | 13,539 | https://gratipay.com/~Sindre%20Sorhus/ |  |
| 2 | 4,750 | https://gratipay.com/~feross/ |  |
| 3 | 1,955 | https://gratipay.com/~bbatsov/ | has three teams |
| 4 | 1,898 | https://gratipay.com/~kangax/ |  |
| 5 | 1,761 | https://gratipay.com/~kytrinyx/ | digest [this history](https://github.com/gratipay/gratipay.com/issues/3481) before contacting |
| 6 | 1,417 | https://gratipay.com/~technomancy/ |  |
| 7 | 1,304 | https://gratipay.com/~sokra/ |  |
| 8 | 1,196 | https://gratipay.com/~IonicaBizau/ |  |
| 9 | 1,175 | https://gratipay.com/~dougwilson/ |  |

**Comment by whit537** [comment]
We should also look through the folks affected by #432, if that's the work we're going to pitch. The folks who already have a Team may have less incentive to endorse us.

**Comment by whit537** [comment]
We also don't want to discount cases where the project is higher-profile than the person, e.g.:

https://gratipay.com/Font-Awesome/
https://gratipay.com/Moment.js/
https://gratipay.com/karma/

A manual sweep through the list may be in order as well, since the short-list should certainly also include:

https://gratipay.com/jsbin/
~= https://gratipay.com/~rem/

**Comment by whit537** [comment]
I mixed star and follower counts into [the spreadsheet](https://docs.google.com/spreadsheets/d/1SAX0yzwbIjvgpUuvbkNXIhxDMwGzIqOmxIHi67TyJu0/edit) for any repo linked in the homepage, onboarding or todo fields, using the script below and some sheet manipulation.

```python
#!/usr/bin/env python -u
from __future__ import absolute_import, division, print_function, unicode_literals

import csv
import os
import re
import requests
import sys


teams = open('teams.txt').read().split()
gratipay = 'https://gratipay.com/'
gratipay_public = gratipay + '{}/public.json'
github_user = 'https://api.github.com/users/{}'
github_repo = 'https://api.github.com/repos/{}'
out = csv.writer(sys.stdout)
USER = os.environ['GH_USER']
PASS = os.environ['GH_PASS']


for team in teams:
    json = requests.get(gratipay_public.format(team)).json()

    homepage = json['homepage']
    onboarding = json['onboarding_url']
    todo = json['todo_url']

    repo = ''
    for link in (homepage, onboarding, todo):
        m = re.match(r'https://github.com/([^/]*/[^/#]*).*', link)
        if not m:
            continue
        repo = m.groups()[0]

    nstars = nfollowers = -1
    if repo:
        nstars = requests.get( github_repo.format(repo)
                             , auth=(USER, PASS)
                              ).json().get('stargazers_count', -2)
        nfollowers = requests.get( github_user.format(repo.split('/')[0])
                                 , auth=(USER, PASS)
                                  ).json().get('followers', -2)
    out.writerow(( gratipay+team+'/'
                 , 'https://github.com/'+repo if repo else ''
                 , nstars
                 , nfollowers
                  ))
```

**Comment by whit537** [comment]
We have 59 Teams with repos with 1,000+ stars, including eight Teams with repos with 10,000+ stars!

| Stars | Team | ~owner | Notes |
|-------:|:---------|:--------|:--------|
| 44,604 | https://gratipay.com/font-awesome/ | ~davegandy |  |
| 27,775 | https://gratipay.com/moment.js/ | ~momentjs |  |
| 18,446 | https://gratipay.com/webpack/ | ~sokra |  |
| 16,465 | https://gratipay.com/md-for-bootstrap/ | ~FezVrasta |  |
| 12,226 | https://gratipay.com/dokku/ | ~josegonzalez | ~progrium (2,745 followers) is also involved |
| 11,348 | https://gratipay.com/webtorrent/ | ~feross |  |
| 11,087 | https://gratipay.com/opencover/ | ~sawilde |  |
| 10,104 | https://gratipay.com/devdocs/ | ~Thibaut |  |

**Comment by whit537** [comment]
Hrm ... some data corruption in the spreadsheet due to dash being sorted differently. One sec ...

**Comment by whit537** [comment]
Okay, s/opencover/opencv/ ...

**Comment by whit537** [comment]
Why isn't https://gratipay.com/karma/ showing up? The Todo link is to the GitHub repo ...

**Comment by whit537** [comment]
Their Todo URL is http instead of https, updating ...

**Comment by whit537** [comment]
I also fixed https://gratipay.com/deadbeef/. That's the only other one I found amongst the -1s on the sheet.

**Comment by whit537** [comment]
I manually updated their numbers on the spreadsheet.

**Comment by whit537** [comment]
Okay! The -2s are fine. I think we have our shortlist:

Per GitHub followers:

1. https://gratipay.com/~Sindre%20Sorhus/  
1. https://gratipay.com/~feross/   
1. https://gratipay.com/~bbatsov/ &larr; three teams!
1. https://gratipay.com/~kangax/   
1. https://gratipay.com/~kytrinyx/ &larr; digest [history](https://github.com/gratipay/gratipay.com/issues/3481) before contacting!
1. https://gratipay.com/~technomancy/  
1. https://gratipay.com/~sokra/    
1. https://gratipay.com/~IonicaBizau/  
1. https://gratipay.com/~dougwilson/
 
Per stars:
 
1. https://gratipay.com/~davegandy  
1. https://gratipay.com/~momentjs   
1. https://gratipay.com/~sokra  
1. https://gratipay.com/~FezVrasta  
1. https://gratipay.com/~josegonzalez
1. https://gratipay.com/~feross 
1. https://gratipay.com/~OpenCV 
1. https://gratipay.com/~Thibaut
 
https://gratipay.com/~progrium &larr; for dokku
https://gratipay.com/~rem &larr; for jsbin

That's 19. Two more things to look at:

 - number of Twitter followers
 - the Teams "Under Review"

**Comment by whit537** [comment]
I've [mixed in](https://docs.google.com/spreadsheets/d/1SAX0yzwbIjvgpUuvbkNXIhxDMwGzIqOmxIHi67TyJu0/edit?usp=sharing) teams under review ([gist](https://gist.github.com/whit537/0d9dae8399b66f830d03689c8157ddf5)). The two that jump out at me that we might add are:

1. https://gratipay.com/mottie/
1. https://gratipay.com/tymondesigns/

**Comment by whit537** [comment]
Twitter followers count suggests we add:

1. https://gratipay.com/~jamescampbell/ (14,246) and maybe
1. https://gratipay.com/~zefhemel/ (8,723)

**Comment by whit537** [comment]
I added a "Score" column, which sums the Twitter followers and Stars with the greater of the Gratipay ~owner and repo owner GitHub followers. It suggests https://gratipay.com/~mxstbr/ (13,511) and perhaps https://gratipay.com/~thomaspark/ (8,541).

**Comment by whit537** [comment]
This listing based primarily on score seems as good a one as any. The next step is to draft an email and send it out!

| Score  | ~user |
|:-------|:--------|
| 71,540 | https://gratipay.com/~davegandy/ |
| 46,458 | https://gratipay.com/~Sindre%20Sorhus/ |
| 28,852 | https://gratipay.com/~momentjs/ |
| 27,213 | https://gratipay.com/~feross/ |
| 25,312 | https://gratipay.com/~kangax/ |
| 19,749 | https://gratipay.com/~sokra/ |
| 18,400 | https://gratipay.com/~OpenCV/ |
| 17,376 | https://gratipay.com/~FezVrasta/ |
| 17,253 | https://gratipay.com/~josegonzalez/ |
| 14,258 | https://gratipay.com/~jamescampbell/ |
| 13,511 | https://gratipay.com/~mxstbr/ |
| 13,311 | https://gratipay.com/~kytrinyx/ |
| 13,272 | https://gratipay.com/~bbatsov/ |
| 11,207 | https://gratipay.com/~Thibaut/ |
| 10,199 | https://gratipay.com/~zefhemel/ |
| 9,626  | https://gratipay.com/~bbatsov/ |
| 8,541  | https://gratipay.com/~thomaspark/ |
| 7,887  | https://gratipay.com/~Dignifiedquire/ |
|        | https://gratipay.com/~rem/ |
|        | https://gratipay.com/~progrium/ |

**Comment by whit537** [comment]
I guess let's use this pad for the email draft:

https://public.etherpad-mozilla.org/p/gratipay-moss-track-2-2016-endorser

**Comment by whit537** [comment]
> We've decided to implement a more batched process for considering MOSS 
applications, so we'll be having an application deadline every 3 months 
for that round of consideration. 
>
The first of these deadlines is 11:59pm UTC on August 31st, 2016 - 
basically, the end of August. They will then occur every three months 
thereafter. 
>
So if you or someone you know is pondering a MOSS application, now is 
the time to send it in. The deadline is a little over a week away. 

https://groups.google.com/forum/#!topic/mozilla.moss/E5d4Owvb7lk 😞

**Comment by whit537** [comment]
That's a curveball. We want to be done with #432 by the time the next deadline will roll around (Dec 31). :-/

**Comment by whit537** [comment]
Also:

> We expect the vast majority of MOSS awards to be singular rather than regular - that is, support for doing a particular defined thing, rather than ongoing support for a project in general.

https://wiki.mozilla.org/MOSS/Mission_Partners

**Comment by whit537** [comment]
Okay, I posted the following on the mailing list. Let's circle back in November.

> Gerv, et al.,
>
I rewrote Gratipay's MOSS Mission Partners application yesterday ... and then learned about the new quarterly deadline ... which we just missed. D'oh! :-) 
>
In case you're interested to see how our application has progressed, here it is: http://bit.ly/2cWuX7b. 
>
If you're willing to consider our application out of band, we'd be happy to submit it. Otherwise we'll check back in late November to see about the next quarter. 
>
Either way, thanks again for running this program! All the best! :-) 

https://groups.google.com/forum/#!topic/mozilla.moss/CZY1TiyZOyY

**Comment by whit537** [comment]
> Applications for Foundational Technology and  Mission Partners remain open, with the next batch deadline being the end of November 2016.

https://blog.mozilla.org/blog/2016/10/03/moss-supports-four-more-open-source-projects-with-300k/

**Comment by whit537** [comment]
> Would you like to be added to https://wiki.mozilla.org/MOSS/Friends? If so, email me. And feel free to tell your members about MOSS.

https://twitter.com/gerv/status/801013422724939776

💃

**Comment by whit537** [comment]
To: Gerv at Mozilla
Subject: Gratipay → MOSS/Friends

>>Thank you for the [invitation](https://twitter.com/gerv/status/801013422724939776) to add Gratipay to MOSS/Friends! :-)
>> 
>>How about the following?
>>
>>**Gratipay** 
>>
>>Gratipay helps companies sustain the open-source ecosystem they depend on, through weekly recurring funding. As of late 2016 there are over 200 projects on Gratipay, which have collectively received more than $1M over the past four years. Gratipay itself is an [open organization](http://inside.gratipay.com/big-picture/welcome), and is [funded](https://gratipay.com/Gratipay/) on Gratipay. [Apply here](https://gratipay.com/new).

---------------------

> Done.

https://wiki.mozilla.org/MOSS/Friends 💃

**Comment by whit537** [comment]
https://twitter.com/Gratipay/status/801030136761982976

**Comment by whit537** [comment]
Maybe we should just go for it?

> We're going up against Open Collective, which has [raised](https://opencollective.com/faq#investors) $805,000 in venture capital. We think open source deserves an open platform, non-profit and funded on itself. That's why we're asking for the full amount of $250,000.

**Comment by whit537** [comment]
I've reached out to someone privately about endorsing us.

**Comment by whit537** [comment]
@teabass has agreed to endorse us! 💃

> Any particular points you'd like me to highlight?

**Comment by andrew** [comment]
👋

**Comment by whit537** [comment]
@andrew has agreed to endorse us! 💃

> Any particular points you'd like me to highlight?

**Comment by whit537** [comment]
@andrew Since you've successfully [gotten](https://medium.com/@teabass/libraries-io-joins-brave-new-software-c8cb7bba93bf) grant funding recently, perhaps you could advise on how to approach this. Do we go for https://github.com/gratipay/inside.gratipay.com/issues/637#issuecomment-262227328, or should we try for something more modest? What do _you_ see as Gratipay's distinctives and needs?

**Comment by whit537** [comment]
> Ask your endorser to write a paragraph or two on why supporting your work via this grant is a great idea.

What future for Gratipay could you get excited about? Are there particular features or improvements you'd like to see? We've got our [own ideas](http://inside.gratipay.com/big-picture/roadmap), of course. Let's find something we can converge on. :)

**Comment by andrew** [comment]
cc'ing @benjam in as he's the grant specialist (although snowed under for the next week so might be slow to respond)

As you mentioned in https://www.youtube.com/watch?v=Rj88uGUB7tw and https://github.com/gratipay/inside.gratipay.com/issues/852#issuecomment-255098337, making it easy for businesses to easily see and fund all the open source software the depend upon across their whole business has a lot of potential to help business realize the risk factor of not funding all to of these projects that their business depends upon, something which until now has been a very hard thing to do.

**Comment by andrew** [comment]
With respect to how to plan the grant, I'd focus on specific outcomes and how they will have an ongoing positive impact as well as what other good initiatives those outcomes will support.

I seem to remember that MOSS prefers to fund specific goals/projects rather than ongoing maintenance although I don't believe that they have rejected any application yet.

**Comment by whit537** [comment]
I do think we need to focus on companies more than projects. There's an obvious felt need among projects to receive funding, signing them up is not the hard part. The hard part is finding the hook to convince companies to sign up and pay up. Making it as easy as possible to get money to the right place in terms of dependencies is one aspect, but that really is just removing barriers, which is not enough to motivate companies to action. To be honest I think diversity is probably the strongest pain point for companies (our experience under Gittipay 1.0 bears this out). Not sure how that plays into a grant application here ...

**Comment by mattbk** [comment]
What dp you mean when you say "diversity"?

**Comment by whit537** [comment]
I mean demographic diversity, primarily gender, race, and class.

**Comment by whit537** [comment]
The demographics in tech are out of whack. Here's the [chart](http://www.npr.org/sections/money/2014/10/17/356944145/episode-576-when-women-stopped-coding) I find myself returning to:

![44d851e4-cb5f-11e4-9e39-2545052d0bae](https://cloud.githubusercontent.com/assets/134455/20528386/00649ea6-b09a-11e6-8cdc-688175c0b24f.png)

**Comment by whit537** [comment]
Alright, I've started dusting off [the application draft](https://public.etherpad-mozilla.org/p/gratipay-moss-track-2-2016). I'm going to spend some time with the [roadmap](http://inside.gratipay.com/big-picture/roadmap) and probably factor out a strategy document. It's encouraging that Gerv [invited](https://twitter.com/gerv/status/801013422724939776) us to join MOSS/Friends. That means we're on their radar and pretty clearly mission-aligned. 👍

**Comment by whit537** [comment]
Here's that [strategy](http://inside.gratipay.com/big-picture/strategy) doc. The [roadmap](http://inside.gratipay.com/big-picture/roadmap) is simpler now, ready for a going-over.

**Comment by whit537** [comment]
Okay, I've reworked [our application](https://public.etherpad-mozilla.org/p/gratipay-moss-track-2-2016).

> Any particular points you'd like me to highlight?

@andrew Do you buy the idea that Gratipay matters because it's more truly open than Open Collective?

**Comment by andrew** [comment]
@whit537 I don't think I have enough information from both parties to make that distinction and that, even if that distinction did exist then it might be damaging to wider open source sustainability efforts to be divisive about it rather than supportive of all efforts to create a more sustainable community.

**Comment by whit537** [comment]
Fair enough. From my perspective, what I am seeing is that Open Collective has $805,000 in [VC funding](https://opencollective.com/faq#investors), and Gratipay has $0 in funding. Gratipay may already be irrecoverably irrelevant. Without funding, we're _definitely_ done.

I actually [talked](https://github.com/gratipay/inside.gratipay.com/issues/458#issuecomment-262354841) with Pia the other day, and asked her what it could look like for those of us working on Gratipay to give up and start helping with Open Collective instead. She basically said, "Sorry, we can't accommodate that right now. Hopefully we can figure out open collaboration before our investors want to cash out 10 years from now." 😞 That feels gross. Not only are they killing Gratipay, but they are also not letting us collaborate with them instead. That doesn't feel like open source to me. I think the world and open source should have a crowdfunding and payments platform that is competitive and relevant ... _and_ open. Gratipay is open, but we're not competitive right now, and we won't be without funding.

> Ask your endorser to write a paragraph or two on why supporting your work via this grant is a great idea.

Can we find a non-divisive way to say that it's "a great idea" for Mozilla to give Gratipay a chance to stay competitive? For better or for worse, it's competition (from Open Collective in particular right now, but there are [others](https://gratipay.news/seven-ways-to-crowdfund-your-open-source-project-f169b42311b2)) that is driving our need for funding.

If Gratipay disappeared, why would it matter? Would you care?

**Comment by whit537** [comment]
> With respect to how to plan the grant, I'd focus on specific outcomes and how they will have an ongoing positive impact as well as what other good initiatives those outcomes will support.

Probably the thing to do is to ignore the intense [burning](https://github.com/gratipay/inside.gratipay.com/issues/900#issuecomment-263020614) sensation that I feel, and pitch them on $100k or whatever to add the "give to a bundle of dependencies" feature that we've been talking about:

> As you mentioned in https://www.youtube.com/watch?v=Rj88uGUB7tw and #852 (comment), making it easy for businesses to easily see and fund all the open source software the depend upon across their whole business has a lot of potential to help business realize the risk factor of not funding all to of these projects that their business depends upon, something which until now has been a very hard thing to do.

**Comment by nobodxbodon** [comment]
@whit537 I'm trying to see the distance between Open Collective and Gratipay here. From their number [here](https://opencollective.com/opensource), seems the total annual budget of all the projects is 100k. Not sure what the 277k budget on their homepage is about. Our weekly donation is around 1k, so annually 50k. It doesn't seem a big distance, considering they have VC to boost their marketing.

**Comment by nobodxbodon** [comment]
> If Gratipay disappeared, why would it matter? Would you care?

@whit537 you can change "Gratipay" to the name of any endangered species :) Maybe you are instead asking "if it hangs in there, would it worth all the work and pain?"

**Comment by mattbk** [comment]
@nobodxbodon, weekly money moved is $1K, but Gratipay receives far less.

**Comment by nobodxbodon** [comment]
@mattbk yes I'm aware of that. I was comparing the platform capability of moving money. IIUC Open Collective doesn't get a cut from donation, and income only comes from VC funds?

**Comment by mattbk** [comment]
@nobodxbodon :bowing_man: 

> Can we find a non-divisive way to say that it's "a great idea" for Mozilla to give Gratipay a chance to compete? 

The non-divisive way of saying this is to say that we don't see the need to compete--because there' room enough for everyone in the marketplace at the moment. See also our previous discussions about the long tail and growth mindset.

**Comment by whit537** [comment]
!m @mattbk @nobodxbodon @andrew ❤️

**Comment by whit537** [comment]
Okay! I've revisited https://github.com/gratipay/inside.gratipay.com/issues/637#issuecomment-236044779, in particular:

> we need to make it _really really easy_ for them to write a sentence of the form "This award will be used to ______________."

... and I've rewritten our [grant application](https://public.etherpad-mozilla.org/p/gratipay-moss-track-2-2016) to focus on concrete outcomes. 👍

@andrew I've copy/pasted your comments from https://github.com/gratipay/inside.gratipay.com/issues/637#issuecomment-262232768 to the "Please share an endorsement from your endorser" section. Good for a first cut, I think. 👍 Does it work for you to further edit that section directly on [the pad](https://public.etherpad-mozilla.org/p/gratipay-moss-track-2-2016)?

I feel much better about this version of the application. I think it gives MOSS something positive and concrete to say "Yes!" to. How is it looking to the rest of yinz?

> [Gratipay](https://gratipay.com/): $250,000. Gratipay is a website that helps companies pay the open source projects they depend on. This award will be used to redesign the site, integrate package managers, and improve operations.

Due in four days!

**Comment by rohitpaulk** [comment]
:) Looks good!

> That's why we're approaching the Mozilla Foundation instead of pursuing venture capital

That isn't entirely true (https://github.com/gratipay/inside.gratipay.com/issues/68)

**Comment by whit537** [comment]
>> That's why we're approaching the Mozilla Foundation instead of pursuing venture capital
>
>That isn't entirely true (#68)

Yeah, my thinking at this point is that if/when we land our first non-profit grant, then we will convert to a non-profit (#916) and close the door on venture capital. Do we need to change what we say in the application here?

**Comment by whit537** [comment]
@timothyfcook and I are workshopping the [application](https://public.etherpad-mozilla.org/p/gratipay-moss-track-2-2016).

**Comment by whit537** [comment]
Initial advice from @timothyfcook (who has a fair bit of experience with grantwriting via @saxifrageschool):

- don't request the max, build up from the pieces rather than starting from the top
- traditional grant-making doesn't like funding operational expenses ("you can have some, but usually only 10-20%")—everything below the top two items is operations

**Comment by whit537** [comment]
> A year or two ago I referred a fairly straightforward website project that ended up costing $65,000.

Another data point is [Zeldman](http://studio.zeldman.com/)'s slider, which defaults to $100,000 on the low end, and bottoms out at $45,000.

![image](https://cloud.githubusercontent.com/assets/134455/20682323/8cd45572-b575-11e6-8cef-c6609578e7d9.png)

**Comment by whit537** [comment]
I think of http://metalab.co/ as the top of the top shelf. They made Slack. If we email them with a budget of $95,000, they will ignore us.

**Comment by whit537** [comment]
Alright, whittled down to $185,000:

> **$95,000** will be used to hire a design firm to refresh the visual design and user experience of Gratipay. Design quality is the [number one factor in website credibility](https://www.nngroup.com/articles/trustworthy-design/), but open source has historically had a reputation for bad design. We're stuck in that rut, and we want to get out of it!
>
>**$60,000** will be distributed to Gratipay collaborators to integrate package managers into Gratipay. Today's open source funding options are geared towards isolated projects. In production, however, companies depend on a long tail of hundreds of open source packages. We want to use package managers to make it easy for companies to see and fund exactly the open source software they depend on.
>
>**$15,000** will be used for operations (customer support, issue tracker maintainance, legal, accounting, etc.).
>
>**$15,000** will be donated from Mozilla through Gratipay to other open source projects that are meaningful to Mozilla. Gratipay is a [friend of MOSS](https://wiki.mozilla.org/MOSS/Friends). Let's #fundopensource together!

The $60,000 is for us, team (plus whatever of the $15,000 for operations we don't spend on lawyers and CPAs). I'm thinking of this as being for the year, for 2017. I think we want to come in solidly under $200,000, because the largest previous gift was $150,000 for Tor. I think if we start to push $200,000 then it will get increasingly harder to say "yes" to us. I also rewrote to work in multiples of 5 instead of 3 (95,000 instead of 96,000, etc.) based on the numbers I am seeing from [round 1](https://blog.mozilla.org/blog/2016/06/22/mozilla-awards-385000-to-open-source-projects-as-part-of-moss-mission-partners-program/).

**Comment by whit537** [comment]
Alright, done drafting for now.

I think the last major piece here is the endorsement. 

@andrew Does it still look like you'll be able to help us out with that? What do you need from me/us? We've recently set up Slack, so if it's easier feel free to drop in there to chat (I've sent you an invite). 🙇

**Comment by whit537** [comment]
> You put a heavy price on design @whit537

@clone1018 in [slack](http://gratipay.slackarchive.io/gratipay/-/1480071845.000126/1480366454.000227/1480365689000219/)

**Comment by nobodxbodon** [comment]
> $95,000 will be used to hire a design firm to refresh the visual design and user experience of Gratipay. Design quality is the number one factor in website credibility, but open source has historically had a reputation for bad design. We're stuck in that rut, and we want to get out of it!

I'm a bit concerned about putting more than half of the fund to UX design. I can't agree more about appeal is indeed very critical, but it sounds like trying our luck with too much to put down. Besides, is there any timeline about this?

> $60,000 will be distributed to Gratipay collaborators to integrate package managers into Gratipay. Today's open source funding options are geared towards isolated projects. In production, however, companies depend on a long tail of hundreds of open source packages. We want to use package managers to make it easy for companies to see and fund exactly the open source software they depend on.

Do you mean npm only, or other package managers as well? If latter, maybe better to make it clear?

**Comment by whit537** [comment]
> I'm a bit concerned about putting more than half of the fund to UX design. I can't agree more about appeal is indeed very critical, but it sounds like trying our luck with too much to put down.

Say more? What do you see as the risks? I think we risk more by budgeting too little, in which case we'll only be able to afford a third-rate firm, and we won't get a quality outcome.

> Besides, is there any timeline about this?

I have in mind six months to a year.

> Do you mean npm only, or other package managers as well? If latter, maybe better to make it clear?

The [plan](https://github.com/gratipay/gratipay.com/issues/4148) is to start with npm, get it working there, and then work with Libraries.io on integrating the rest of the package managers based on our experience with what we needed for npm. It's hard to say how far $60,000 will get us down that path. Do you think as it's written it is specific to npm?

**Comment by clone1018** [comment]
A little more discussion here: http://gratipay.slackarchive.io/gratipay/-/1480095149.000138/1480368037.00024/1480367790000230/ (scroll up)

**Comment by nobodxbodon** [comment]
> What do you see as the risks? I think we risk more by budgeting too little, in which case we'll only be able to afford a third-rate firm, and we won't get a quality outcome.

I agree in general that you get what you pay for. It's mostly about the way we put in the application, I suppose. With "to hire a design firm to" it sounds like a decision without due research or evaluation. Maybe it's better to have some references?

> I have in mind six months to a year.

Shall we put in application to make the plan more concrete?

> Do you think as it's written it is specific to npm?

Actually, I would have not fully understood the term "package manager" here if I hadn't known the [plan](https://github.com/gratipay/gratipay.com/issues/4148) beforehand, so I suppose the reviewer of the application may feel the same. So maybe it's better to illustrate a little more?

**Comment by whit537** [comment]
> So maybe it's better to illustrate a little more?

I'm getting some private rumblings from a lurker that more specificity is indeed called for. Hopefully detail is forthcoming ... ? ;-)

**Comment by bbangert** [comment]
As @whit537 requested it here, I noted that for another applicants MOSS approval that I worked with, when they requested money to implement a feature, MOSS wanted a fairly detailed set of answers to questions like:

 - Who and how many developers will be making the feature, how many
hours estimated at what rate?
- What exactly will be done upon completion? What does the finished
'thing' look like?
- Divide the work into time periods so that progress can be gauged
throughout the year. (They do check-in on the projects to see what
progress is being made, how it lines up to the original estimate, etc)

If 'package managers' is not a very well defined thing, with a
time-line, they'd likely either reject it or come back to you
requiring more details. This is part of why I'm asking so many
questions about the package manager thing, because the scope of work
changes radically depending on the answers, and they've asked in the past for a defined feature set to justify how much money is needed to
implement it.

That was for a Foundational Technology MOSS grant though, not a Mission Partners one. Though I would still expect them to want a broad overview along those lines to gauge whether the grant was delivered on or not. It's entirely possible that a similar breakdown will be desired regarding the website redesign, or at least whether this quote is just made-up or was actually quoted by a design firm for a specified scope of work.

**Comment by whit537** [comment]
Thanks, @bbangert. :-)

**Comment by clone1018** [comment]
So we just need a plan. It will be a roadmap for Gratipay and information for potential funding sources.

Right?

**Comment by whit537** [comment]
> So we just need a plan.

Alright, so let's delegate. Who is going to work on which part of the plan? Are we agreed on the overall structure of the grant? Namely ...

1. site redesign
1. to-be-defined "package manager" feature
1. a bit for operations
1. a bit for giving through Gratipay

**Comment by whit537** [comment]
"Grant folks will tell you what details they want. And if they don't ask for them, that means they don't have the time to read about them and they don't think they're that important." —@timothyfcook

I'm going to compare the Mission Partners application and the Foundational Tech applications ...

**Comment by whit537** [comment]
[Foundational Technology](https://docs.google.com/forms/d/e/1FAIpQLSdSSJ2jCuNCSY2SlIReOPwuF-ILnkt7IpNk4YkLKPn1spJoMA/viewform)

[Mission Partners](https://docs.google.com/forms/d/e/1FAIpQLSe9a9h4afnJnDp3rRRHB4i1Xnb3otzfaQrb9uXVeoeXlB3NSg/viewform)

**Comment by whit537** [comment]
Same:

FT:

> **What are the concrete, specific outputs and outcomes this grant would produce?** *
> Please describe what you would use the funds for - what you are going to build, hack or fix, and how that would be of benefit to the world. (Max 8k chars)

MP:

> **What are the concrete, specific outputs and outcomes this grant would produce, and how do those activities further the Mozilla Mission?** *
> Please describe what you would use the funds for - what you are going to build, hack or fix. Also, explains how it furthers our mission, perhaps tying your activities to items in the Manifesto - https://www.mozilla.org/about/manifesto/. (Max 8k chars)

**Comment by whit537** [comment]
Alright, so I take it that @bbangert is a "grant folk" for our purposes here, and he has told us what "they" want. :-)

Questions therefore stand:

1. Are we agreed on the overall structure of the grant? Namely ...
 1. site redesign
 1. to-be-defined "package manager" feature
 1. a bit for operations
 1. a bit for giving through Gratipay
1. Who is going to work on which part of the plan?

**Comment by andrew** [comment]
@whit537 I think there may be some conflict of interest with me endorsing the grant was well as providing part of the tooling to support the "package manager" feature, as we're also planning on applying for MOSS funding at some point I wouldn't want to jeopardize that

**Comment by whit537** [comment]
> there may be some conflict of interest with me endorsing the grant was well as providing part of the tooling to support the "package manager" feature

Ah, well. I wondered about that. Okay! Thanks anyway, @andrew. :-) Feel free to unsubscribe to avoid further notifications from this ticket, I imagine we're going to be generating quite a few over the coming days. See you around! :-)

**Comment by whit537** [comment]
... aaaaaaaaaaaaaaand we're back to no endorser. :o)

@bbangert Think there's a chance we can converge on a scope of work for the package manager integration that you'd feel good about endorsing? :)

Let's dig into that for a minute ...

**Comment by whit537** [comment]
The fundamental idea is to get a viral loop going again like we had under Gittipay 1.0. As @clone1018 [points out](http://gratipay.slackarchive.io/gratipay/-/1480336816.000159/1480370112.000261/1480367790000230/), that happened with a not-so-fancy visual design, suggesting that we could remove that piece from this grant without detracting from the fundamental idea. At the very least, we could plow it in with feature development rather than itemizing it separately.

Here's the way Gittip worked:

- We had pages on Gittip representing social media accounts, `/on/github/bbangert/`, `/on/twitter/oprah/`, etc.
- You could "pledge" to any of these "accounts elsewhere." A pledge was just that: a promise to give money _if_ the person accepted it by joining Gittip. This created incentive for people to join: free money!
- Once you signed up, you would put a badge on your README, and/or a widget on your blog. Now you're promoting the network by promoting yourself.
- Top givers and receivers were on the homepage.
- We _really_ started taking off when you could give charitably to diversity activists *and* more pragmatically to open source projects.

Basically I think we need to reimplement that, but for projects instead of individuals.

- Have pages on Gratipay representing open source packages, `/on/npm/express/`, `/on/pypi/requests/`, etc. (https://github.com/gratipay/gratipay.com/issues/4148)
- Bring back pledging (https://github.com/gratipay/gratipay.com/issues/4195).
- Bring back widgets, and/or README integration like OC has.
- Bring back top givers (and receivers?) on the homepage (https://github.com/gratipay/gratipay.com/issues/2695).
- Bring back a charitable element (https://github.com/gratipay/gratipay.com/issues/4209).

I do think there's a role for package manager integration (more on that at https://github.com/gratipay/gratipay.com/pull/4135#issuecomment-263393515), but the bigger picture is bigger. How do we break that down into a grant application?

**Comment by bbangert** [comment]
@whit537 I'm a bit skeptical in general about funding a platform for funding OSS developers vs. funding OSS dev's for work directly on a specific project, mainly for the reasons that follow.

I read articles like this (http://www.infoworld.com/article/3144546/security/time-is-running-out-for-ntp.html) and I'll admit they reinforce my thinking that the entire concept of funding OSS via donations is totally broken. Maybe companies will donate to trendy tech they use, but then they skimp on core tech that everyone requires (in this case NTP, but see-also OpenSSL)? How is this good for the Internet or OSS as a whole?

Perhaps its the car broken on the side of the freeway problem. The one where a car broken down on the side of a heavily-trafficked freeway can have a greater delay in having a motorist stop to help than on a very rarely-traveled road.... because on the heavy-traffic route everyone assumes someone else will stop to help out while on the rarely-traveled road the passing motorists believes (rightfully so) that it's critical they stop.

Are the OpenSSL and NTP projects on Gratipay? Does it seem likely that if a company submits their package manifesto it will include such core, critical, and under-funded things such as these? Or will they just assume someone else is dealing with that and throw money at the latest fad OSS library instead of the old, boring, and legacy thing everyone actually is using?

How does Gratipay help provide feedback to companies about what projects are over or under-funded, and whether the resources are actually urgently needed? Could Gratiay provide a way for a project to indicate whether developers are willing or not to work full-time on security or bug fixes such that funding it would result in substantially better results than a OSS lib where the dev is going to treat it as merely some extra dough?

I mean, I think it's very useful to be honest in this discussion. In that, I am grateful that @whit537 has been exceptionally transparent in his handling of all of this. I'd love to see such honesty from projects requesting money on whether any of them would actually quit a day-job to survive purely on donations (I can imagine how tough such a choice would be).

So.... to stop my rant here.. what would I feel good about endorsing? I think the package manager concept has some good foundational underpinnings, but if someone is ready to commit money to OSS, is there an opportunity to do better? Do they use NTP/OpenSSL or some other severely under-funded OSS project that is critical to the Internet that they just didn't know needed a bit more money? Maybe right then might be a good time to mention how important it is that some of their donation be directed there?

If this feature was about surfacing OSS projects that are critical to wide swaths of the Internet that apparently go under-funded until an article like that appears (I sincerely hope an article like that results in some funding, but maybe heavy-freeway syndrome will hit again)... I'd be behind it in a second. :)

Let's take the NTP example. How could such a gratipay page for this project be setup? I would expect to see whether or not the funds go to developers attempting to make this a full-time job vs. those just appreciating some extra money. It'd be even better for a project like this to see how they figure more funds might increase response-time to handle critical security bugs. Maybe the project could indicate if its a core-req for the majority of linux distros (how many other core libs/daemons need funding besides NTP/OpenSSL?).

Can projects set a spending-goal such that companies/people donating know they can stop when its been accomplished? Or maybe they can see that they've ensured the project is financially sound for the next N years with M developers fully funded? I have yet to see any way to do such a thing.

Handling any/most of these concerns of mine would make me feel quite good about endorsing this.

**Comment by andrew** [comment]
> If this feature was about surfacing OSS projects that are critical to wide swaths of the Internet that apparently go under-funded until an article like that appears (I sincerely hope an article like that results in some funding, but maybe heavy-freeway syndrome will hit again)... I'd be behind it in a second. :) 

@bbangert we'll have to get you to endorse the https://libraries.io grant next year 😉

**Comment by nobodxbodon** [comment]
> Maybe the project could indicate if its a core-req for the majority of linux distros (how many other core libs/daemons need funding besides NTP/OpenSSL?).

@bbangert Great point. This may need a huge dependency graph of all the projects/libraries, with the information about the organizations owning those. I imagine that will require big effort, like @whit537 mentioned above ("the bigger picture is bigger"), and can only be achieved incrementally. [Current effort of integrating npm package manager](https://github.com/gratipay/gratipay.com/issues/4148) could be the first step.

 > Can projects set a spending-goal such that companies/people donating know they can stop when its been accomplished?

Sounds like kickstarter. Not sure if it can be accomplished with the grant, along with other critical features.

**Comment by BenJam** [comment]
Hey

I’ve been lurking too.

I would just like to say that I agree with much of what Ben says here. When posed with the hypothetical question ‘how much of my money do you want in order to “save” FOSS?’ the answer is pretty much always ‘all of it’ which is why we (by which I mean Libraries.io <http://libraries.io/>, the Core Infrastructure team, the Linux Foundation etc) are trying to tackle this problem by prioritising. 

But I do think that there is a way to harness the ‘next fad’ syndrome in OSS: by funding the complete ‘toolchain’ for that new thing roughly inversely proportionate to the value it inherits from its stack. This will over time result in well funded core FOSS and yet sustain that new thing at a rate that enables it to flourish.

rant/

I think the core principle of funding the total ecosystem for a given application, framework or library is sound and that these kinds of initiatives will help provide a more sustainable ecosystem for FOSS in general. At the same time I think that it’s necessary for everyone to work together in this space, regardless of financial, organisational or legal structure. 

Divisive politics is getting us nowhere at the moment… how about we don’t replicate that here?

/rant

B

**Comment by whit537** [comment]
@bbangert et al.—Awesome discussion. 👍

What about a 5% "tax" that goes to core tech? Currently the full face value of a payment goes to the recipient. So if I give $100 to Hip Project, they get all $100. If we implement a tax, Hip would get $95, and $5 would go to core tech.

We'd have a board to decide what receivers qualify as "core tech" (Core Infrastructure Initiative, Internet Bug Bounty, etc.) and how that budget gets split. P.S. I think Gratipay would qualify, since we don't have a hard fee for ourselves otherwise.

To avoid some of the loss of agency involved in a tax, we could grant givers control over their allocation if they want it. In other words, payments directly to qualifying core tech projects would offset one's tax burden.

P.S. I would also want a 5% tax for demographic diversity programs with a board of its own.

**Comment by mattbk** [comment]
> Perhaps its the car broken on the side of the freeway problem. The one where a car broken down on the side of a heavily-trafficked freeway can have a greater delay in having a motorist stop to help than on a very rarely-traveled road.... because on the heavy-traffic route everyone assumes someone else will stop to help out while on the rarely-traveled road the passing motorists believes (rightfully so) that it's critical they stop.

The technical term for this is the [bystander effect](https://en.wikipedia.org/wiki/Bystander_effect) (sorry, married to a psychology professor).

**Comment by whit537** [comment]
> The technical term for this is the bystander effect

Let's [try](http://inside.gratipay.com/howto/deal-with-bad-behavior) to be aware of and work to resist it! ;-)

**Comment by mattbk** [comment]
> Can projects set a spending-goal such that companies/people donating know they can stop when its been accomplished?

Before Gratipocalypse, there was a "goal" feature that would show how close a ~user was to reaching a defined weekly income. I think this would be worth resurrecting, if only to focus on the "sustainability" part--regardless of anything else, I think the goal of Gratipay is for funding to remain constant, rather than require repeated push for funds, grants, or Kickstarter-like campaigns. People (givers) like to see that they're moving the needle. Yes, they'll be giving that amount _every week_, but it drives home the message that the work doesn't stop just because the Indiegogo campaign is over.

**Comment by whit537** [comment]
Deadline is past.

**Comment by JessaWitzel** [comment]
Reopening for next deadline.

**Comment by whit537** [comment]
(per [slack](http://gratipay.slackarchive.io/gratipay/-/1480708956.000947/1480713273.001049/1480712815001036/))
